Source/Mock.cs

```diff
@@ -1206,8 +1207,9 @@ internal object GetDefaultValue(MethodInfo method, DefaultValueProvider useAlter
 			}
 
 			var result = (useAlternateProvider ?? this.DefaultValueProvider).GetDefaultReturnValue(method, this);
+			var unwrappedResult = TryUnwrapResultFromCompletedTaskRecursively(result);
```

Comment:
For the standard default value providers (`Empty` and `Mock`), it would be great if the wrapping (inside the provider) and unwrapping (here) could be avoided. It should be possible to add a fast path as follows:

1. Check whether `this.DefaultValue != DefaultValue.Custom`. If false, stop.
2. Check whether `method` returns a task. If so, ask the provider not for the actual task return type via `GetDefaultReturnValue`, but for the type of the wrapped value via `GetDefaultValue`.
3. Perform the wrapping into completed tasks here, given the value returned from the provider.

This would save us one unnecessary recursive task unwrapping operation.